types/src/v0/impls/header.rs

```diff
@@ -358,10 +358,22 @@ impl Header {
     }
 
     /// Fee paid by the block builder
-    pub fn fee_info(&self) -> FeeInfo {
-        *field!(self.fee_info)
+    pub fn fee_info(&self) -> Vec<FeeInfo> {
+        match self {
+            Self::V1(fields) => vec![fields.fee_info],
+            Self::V2(fields) => vec![fields.fee_info],
+            Self::V3(fields) => fields.fee_info.clone(),
+        }
```

Comment:
We need some way to distinguish bid fees from sequencing fees, likely using an enum.  I'm not sure where the best place to put this information is, but it needs to go somewhere :)